### PR TITLE
Fix cl_show_others_alpha not used in places where it can be

### LINKED ISSUE #6520: Make extras invisible with cl_show_others_alpha 0
Hello, is it possible to make all extras invisible when using cl_show_others_alpha 0?
Like strong/weak hook, particles etc (see gif).

### PR DESCRIPTION
**Pull request by edg-l**
There are some things like damage indicator, spawn particles, etc which we have no way to properly set alpha based on team/solo knowledge because they don't have any info about that. (to my knowledge)

Fixes #6520

## Checklist

- [x] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

File: src/game/client/components/damageind.cpp [hunk 1]
```diff
@@ -5,6 +5,7 @@
 #include <game/generated/client_data.h>
 #include <game/generated/protocol.h>
 
+#include "base/color.h"
```
Comment (Robyt3) [inline src/game/client/components/damageind.cpp:8]: Use `<...>`

File: src/game/client/components/damageind.h [hunk 2]
```diff
@@ -2,6 +2,7 @@
 /* If you are missing that file, acquire a complete release at teeworlds.com.                */
 #ifndef GAME_CLIENT_COMPONENTS_DAMAGEIND_H
 #define GAME_CLIENT_COMPONENTS_DAMAGEIND_H
+#include "base/color.h"
```
Comment (Robyt3) [inline src/game/client/components/damageind.h:5]: Use `<...>`

File: src/game/client/components/nameplates.cpp [hunk 3]
```diff
@@ -221,6 +231,16 @@ void CNamePlates::RenderNameplatePos(vec2 Position, const CNetObj_PlayerInfo *pP
 						StrongWeakStatusColor = color_cast<ColorRGBA>(ColorHSLA(41131));
 						StrongWeakSpriteID = SPRITE_HOOK_WEAK;
 					}
+
+					float a = 1;
```
Comment (Robyt3) [inline src/game/client/components/nameplates.cpp:235]: Maybe rename the variable `a` so it's more clear. There also is an `Alpha` variable already.

File: src/game/client/components/effects.h [hunk 4]
```diff
@@ -18,16 +18,16 @@ class CEffects : public CComponent
...
+	void DamageIndicator(vec2 Pos, vec2 Dir, float Alpha);
 	void ResetDamageIndicator();
-	void PlayerSpawn(vec2 Pos);
-	void PlayerDeath(vec2 Pos, int ClientID);
-	void PowerupShine(vec2 Pos, vec2 Size);
-	void FreezingFlakes(vec2 Pos, vec2 Size);
+	void PlayerSpawn(vec2 Pos, float Alpha);
+	void PlayerDeath(vec2 Pos, int ClientID, float Alpha);
+	void PowerupShine(vec2 Pos, vec2 Size, float Alpha);
+	void FreezingFlakes(vec2 Pos, vec2 Size, float Alpha);
```
Comment (Robyt3) [inline src/game/client/components/effects.h:30]: Add `Alpha = 1.0f` as default parameter so less existing calls need to be changed?

Comment (edg-l) [reply src/game/client/components/effects.h:30]: i prefer to be explicit here (and tbh default params are evil), so its hard to miss, but can change

**Comment by edg-l** [comment]
done

**Comment by edg-l** [comment]
fixed the opacity for jumps, was wrong before

**Pull request merged**